src/selfplay/loop.cc

```diff
@@ -123,13 +123,56 @@ void SelfPlayLoop::CmdSetOption(const std::string& name,
 }
 
 void SelfPlayLoop::SendTournament(const TournamentInfo& info) {
+  const int winp1 = info.results[0][0] + info.results[0][1];
+  const int loosep1 = info.results[2][0] + info.results[2][1];
+  const int draws = info.results[1][0] + info.results[1][1];
+
+  // Initialize variables
+  float percentage = -1;
+  float elo = 10000;
+  float los = 10000;
+
+  // Only caculate percentage if any games at all (avoid divide by 0).
+  if ((winp1 + loosep1 + draws) > 0)
+    percentage =
+        (static_cast<float>(draws) / 2 + winp1) / (winp1 + loosep1 + draws);
+
+  // Calculate elo and los if percentage strictly between 0 and 1 (avoids divide
+  // by 0 or overflow).
+  if ((percentage < 1) && (percentage > 0))
+    elo = -400 * log(1 / percentage - 1) / log(10);
+  if ((winp1 + loosep1) > 0)
+    los = .5 +
+          .5 * std::erf((winp1 - loosep1) / std::sqrt(2.0 * (winp1 + loosep1)));
+
   std::string res = "tournamentstatus";
```

Comment:
I think it's cleaner to create `std::ostringstream res` instead of res, and output to it, and then just return `res.str()` from the function.

Reply:
its just like it was before. If you really insist i can change it though.

Reply:
Alternatively, you can have a helper function:
```cpp
std::string FormatFixed(double num, int width, int precision);
```
and call it whereever needed.
But current way of mixing two techniques is kind of untidy.

Reply:
done